packages/flutter_tools/lib/src/native_assets.dart

```diff
@@ -415,3 +437,203 @@ Uri buildUriMultiple(Uri projectUri) {
   final String buildDir = build_info.getBuildDirectory();
   return projectUri.resolve('$buildDir/native_assets/multiple/');
 }
+
+
+/// Dry run the native builds.
+///
+/// This does not build native assets, it only simulates what the final paths
+/// of all assets will be so that this can be embedded in the kernel file.
+Future<Uri?> dryRunNativeAssetsLinuxWindows({
+  required NativeAssetsBuildRunner buildRunner,
+  required Uri projectUri,
+  bool flutterTester = false,
+  required FileSystem fileSystem,
+  required OS os,
+}) async {
+  if (await hasNoPackageConfig(buildRunner) || await isDisabledAndNoNativeAssets(buildRunner)) {
+    return null;
+  }
+
+  final Uri buildUri_ = nativeAssetsBuildUri(projectUri, os);
+  final Iterable<Asset> nativeAssetPaths = await dryRunNativeAssetsLinuxWindowsInternal(
+    fileSystem,
+    projectUri,
+    flutterTester,
+    buildRunner,
+  );
+  final Uri nativeAssetsUri = await writeNativeAssetsYaml(
+    nativeAssetPaths,
+    buildUri_,
+    fileSystem,
+  );
+  return nativeAssetsUri;
+}
+
+Future<Iterable<Asset>> dryRunNativeAssetsLinuxWindowsInternal(
+  FileSystem fileSystem,
+  Uri projectUri,
+  bool flutterTester,
+  NativeAssetsBuildRunner buildRunner,
+) async {
+  const OS targetOs = OS.linux;
```

Comment:
This is incorrect, at least for the logging (and if for the code, very confusing, so needs a comment).

Reply:
This also suggests there isn't testing of the logging happening for each platform.